apps/google-analytics-4/frontend/src/helpers/contentfulContext.ts

```diff
@@ -0,0 +1,31 @@
+import { ContentfulContext } from '../types';
+import { KnownSDK, locations } from '@contentful/app-sdk';
+
+const Locations = [
+  locations.LOCATION_APP_CONFIG,
+  locations.LOCATION_ENTRY_FIELD,
+  locations.LOCATION_ENTRY_EDITOR,
+  locations.LOCATION_DIALOG,
+  locations.LOCATION_ENTRY_SIDEBAR,
+  locations.LOCATION_PAGE,
+  locations.LOCATION_HOME,
+];
+
+export const location = (sdk: KnownSDK) => {
+  let currentLocation = undefined;
+
+  Locations.forEach((location) => {
```

Comment:
i'm not super opinionated here but if we're doing a procedural loop in wonder if `for...in` is a better choice?

Reply:
This is going to cause a bug Im pretty sure. In a `forEach`, when you return, you dont return out of the entire function (similar how a .map return will return out of the current context but not the context of the entire function.

You can use the `.find()` higher order array function.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Aka. 

```
return Locations.find((location) => sdk.location.is(location))
```

This will either return undefined (when it can't find it) or the matching object that fulfills the condition as true

Reply:
Another benefit is that it turns all of this logic into one line :)

Reply:
refactored to just be: ```Locations.find((location) => sdk.location.is(location))``` – i was being an idiot :D